File: ClearAzureQueues/Models/AccountSelectionModel.cs
using System.Collections.ObjectModel;
using System.Windows;
using System.Linq;

namespace ClearAzureQueues.Models {

    public class AccountSelectionSettings {
        public QueueSelectionSettings[] Accounts { get; set; }
    }

    public class AccountSelectionModel : DependencyObject {

        public AccountSelectionSettings Settings {
            get {
                return new AccountSelectionSettings() {
                    Accounts = Accounts.Select(x => x.Settings).ToArray()
                };
            }
        }

        public static readonly DependencyProperty AccountsProperty =
            DependencyProperty.Register("Accounts", typeof(ObservableCollection<QueueSelectionModel>), typeof(AccountSelectionModel));
        public ObservableCollection<QueueSelectionModel> Accounts {
            get { return (ObservableCollection<QueueSelectionModel>)GetValue(AccountsProperty); }
            set { SetValue(AccountsProperty, value); }
        }

        public AccountSelectionModel() {
            Accounts = new ObservableCollection<QueueSelectionModel>();
        }

        public AccountSelectionModel(AccountSelectionSettings settings) {
            Accounts = new ObservableCollection<QueueSelectionModel>(
                settings.Accounts.Select(x => new QueueSelectionModel(x)));
        }
    }
}

File: ClearAzureQueues/Models/AccountSelectionModel.fs
namespace ClearAzureQueues.Models

open System.Collections.ObjectModel
open System.Linq
open System.Windows

open ClearAzureQueues
open ClearAzureQueues.Settings

type AccountSelectionModel() =
    inherit DependencyObject()

    let accounts = new ObservableCollection<QueueSelectionModel>()

    static let selectedAccount =
        DependencyProperty.Register("SelectedAccount", typeof<QueueSelectionModel>, typeof<AccountSelectionModel>)

    new { Accounts = accounts } as x =
        AccountSelectionModel() then
            for account in accounts do
                x.Accounts.Add(new QueueSelectionModel(account))

    member public x.Settings =
        { Accounts = x.Accounts |> Seq.map (fun x -> x.Settings) |> Seq.toArray }

    member public x.SelectedAccount
        with get() = match x.GetValue(selectedAccount) with :? QueueSelectionModel as m -> m | _ -> null
        and set(value:QueueSelectionModel) = x.SetValue(selectedAccount, value)

    member public x.Accounts : ObservableCollection<QueueSelectionModel> = accounts

